### DISCUSSION TITLE
Upload maven packages using SNAPSHOT version and classifier

### DISCUSSION THREAD

**Discussion by GabRayz** [community/community#25668] [Actions]
<p>Hi, I’ve been trying to upload a maven artefact to GitHub Packages within a workflow, but encountered a HTTP response 400: Bad request. After some research, I found that it was due to the upload of a SNAPSHOT version, and the use of a classifier (with sources and javadoc). I found a similar issue here: <a href="https://github.community/t/error-on-github-packages-for-snapshot-maven-artifacts-with-a-classifier/116732" class="inline-onebox">Error on Github Packages for `-SNAPSHOT` maven artifacts with a classifier</a>, which isn’t solved. However I found another related issue (<a href="https://github.community/t/partial-support-for-maven-snapshots/115628" class="inline-onebox">Partial support for maven SNAPSHOTs</a>) where jscandale points out that it should be fixed, and where the original poster confirmed it worked, though I still have the problem on my side.<br>
I continued my research but didn’t find any more information that could help me fix the issue.</p>
<p>Here’s my action file</p>
<pre><code class="lang-auto"># This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
# For more information see: https://github.com/actions/setup-java#apache-maven-with-a-settings-path

name: Publish maven package to GitHub Packages

on:
  push:
    branches: [master, sql]
  release:
    types: [created]

jobs:
  build:

    runs-on: ubuntu-latest
    permissions:
      contents: read
      packages: write

    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK 16
        uses: actions/setup-java@v2
        with:
          java-version: '16'
          distribution: 'adopt'
          server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
          settings-path: ${{ github.workspace }} # location for the settings.xml file

      - name: Build with Maven
        run: mvn -B package --file pom.xml

      - name: Publish to GitHub Packages Apache Maven
        run: mvn deploy -s $GITHUB_WORKSPACE/settings.xml
        env:
          GITHUB_TOKEN: ${{ github.token }}

      - uses: actions/upload-artifact@v2
        with:
          name: tesseractLib
          path: |
            /home/runner/work/TesseractLib/TesseractLib/target/*.jar
            !/home/runner/work/TesseractLib/TesseractLib/target/*sources.jar
            !/home/runner/work/TesseractLib/TesseractLib/target/original*.jar
</code></pre>
<p>And my pom.xml file</p>
<pre><code class="lang-auto">&lt;?xml version="1.0" encoding="UTF-8"?&gt;
&lt;project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"&gt;
    &lt;modelVersion&gt;4.0.0&lt;/modelVersion&gt;

    &lt;groupId&gt;onl.tesseract&lt;/groupId&gt;
    &lt;artifactId&gt;tesseractlib&lt;/artifactId&gt;
    &lt;version&gt;2.0-SNAPSHOT&lt;/version&gt;
    &lt;packaging&gt;jar&lt;/packaging&gt;

    &lt;properties&gt;
        &lt;java.version&gt;16&lt;/java.version&gt;
        &lt;maven.compiler.source&gt;16&lt;/maven.compiler.source&gt;
        &lt;maven.compiler.target&gt;16&lt;/maven.compiler.target&gt;
        &lt;project.build.sourceEncoding&gt;UTF-8&lt;/project.build.sourceEncoding&gt;
    &lt;/properties&gt;
         
    &lt;distributionManagement&gt;
         &lt;repository&gt;
             &lt;id&gt;github&lt;/id&gt;
             &lt;name&gt;GitHub Packages&lt;/name&gt;
             &lt;url&gt;https://maven.pkg.github.com/TesseractFR/TesseractLib&lt;/url&gt;
         &lt;/repository&gt;
     &lt;/distributionManagement&gt;
         
    &lt;build&gt;
        &lt;plugins&gt;
            &lt;plugin&gt;
                &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
                &lt;artifactId&gt;maven-compiler-plugin&lt;/artifactId&gt;
                &lt;version&gt;3.8.1&lt;/version&gt;
                &lt;configuration&gt;
                    &lt;source&gt;16&lt;/source&gt;
                    &lt;target&gt;16&lt;/target&gt;
                &lt;/configuration&gt;
            &lt;/plugin&gt;
            &lt;plugin&gt;
                &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
                &lt;artifactId&gt;maven-shade-plugin&lt;/artifactId&gt;
                &lt;version&gt;3.2.4&lt;/version&gt;
                &lt;executions&gt;
                    &lt;execution&gt;
                        &lt;phase&gt;package&lt;/phase&gt;
                        &lt;goals&gt;
                            &lt;goal&gt;shade&lt;/goal&gt;
                        &lt;/goals&gt;
                        &lt;configuration&gt;
                            &lt;createDependencyReducedPom&gt;false&lt;/createDependencyReducedPom&gt;
                        &lt;/configuration&gt;
                    &lt;/execution&gt;
                &lt;/executions&gt;
            &lt;/plugin&gt;
            &lt;plugin&gt;
                &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
                &lt;artifactId&gt;maven-source-plugin&lt;/artifactId&gt;
                &lt;version&gt;3.2.1&lt;/version&gt;
                &lt;executions&gt;
                    &lt;execution&gt;
                        &lt;id&gt;attach-sources&lt;/id&gt;
                        &lt;phase&gt;verify&lt;/phase&gt;
                        &lt;goals&gt;
                            &lt;goal&gt;jar-no-fork&lt;/goal&gt;
                        &lt;/goals&gt;
                    &lt;/execution&gt;
                &lt;/executions&gt;
            &lt;/plugin&gt;
        &lt;/plugins&gt;
        &lt;resources&gt;
            &lt;resource&gt;
                &lt;directory&gt;src/main/resources&lt;/directory&gt;
                &lt;filtering&gt;true&lt;/filtering&gt;
            &lt;/resource&gt;
        &lt;/resources&gt;
    &lt;/build&gt;

    &lt;repositories&gt;
        &lt;repository&gt;
            &lt;id&gt;papermc&lt;/id&gt;
            &lt;url&gt;https://papermc.io/repo/repository/maven-public/&lt;/url&gt;
        &lt;/repository&gt;
    &lt;/repositories&gt;

    &lt;dependencies&gt;
        &lt;dependency&gt;
            &lt;groupId&gt;io.papermc.paper&lt;/groupId&gt;
            &lt;artifactId&gt;paper-api&lt;/artifactId&gt;
            &lt;version&gt;1.17-R0.1-SNAPSHOT&lt;/version&gt;
            &lt;scope&gt;provided&lt;/scope&gt;
        &lt;/dependency&gt;
        &lt;dependency&gt;
            &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
            &lt;artifactId&gt;maven-source-plugin&lt;/artifactId&gt;
            &lt;version&gt;3.2.1&lt;/version&gt;
            &lt;type&gt;maven-plugin&lt;/type&gt;
        &lt;/dependency&gt;
    &lt;/dependencies&gt;
&lt;/project&gt;

</code></pre>
<p>Any help would be appreciated</p>

**Comment by GabRayz** [comment, marked as answer]
<p>After deleting and recreating the package, it works. I can now upload several snapshot versions !</p>

**Comment by yogurtearl** [comment]
related: https://github.com/gradle/gradle/issues/13330

**Comment by ghost** [comment]
> <p>Hi, I’ve been trying to upload a maven artefact to GitHub Packages within a workflow, but encountered a HTTP response 400: Bad request. After some research, I found that it was due to the upload of a SNAPSHOT version, and the use of a classifier (with sources and javadoc). I found a similar issue here: <a href="https://github.community/t/error-on-github-packages-for-snapshot-maven-artifacts-with-a-classifier/116732" class="inline-onebox">Error on Github Packages for `-SNAPSHOT` maven artifacts with a classifier</a>, which isn’t solved. However I found another related issue (<a href="https://github.community/t/partial-support-for-maven-snapshots/115628" class="inline-onebox">Partial support for maven SNAPSHOTs</a>) where jscandale points out that it should be fixed, and where the original poster confirmed it worked, though I still have the problem on my side.<br>
> I continued my research but didn’t find any more information that could help me fix the issue.</p>
> <p>Here’s my action file</p>
> <pre><code class="lang-auto"># This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
> # For more information see: https://github.com/actions/setup-java#apache-maven-with-a-settings-path
> 
> name: Publish maven package to GitHub Packages
> 
> on:
>   push:
>     branches: [master, sql]
>   release:
>     types: [created]
> 
> jobs:
>   build:
> 
>     runs-on: ubuntu-latest
>     permissions:
>       contents: read
>       packages: write
> 
>     steps:
>       - uses: actions/checkout@v2
>       - name: Set up JDK 16
>         uses: actions/setup-java@v2
>         with:
>           java-version: '16'
>           distribution: 'adopt'
>           server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
>           settings-path: ${{ github.workspace }} # location for the settings.xml file
> 
>       - name: Build with Maven
>         run: mvn -B package --file pom.xml
> 
>       - name: Publish to GitHub Packages Apache Maven
>         run: mvn deploy -s $GITHUB_WORKSPACE/settings.xml
>         env:
>           GITHUB_TOKEN: ${{ github.token }}
> 
>       - uses: actions/upload-artifact@v2
>         with:
>           name: tesseractLib
>           path: |
>             /home/runner/work/TesseractLib/TesseractLib/target/*.jar
>             !/home/runner/work/TesseractLib/TesseractLib/target/*sources.jar
>             !/home/runner/work/TesseractLib/TesseractLib/target/original*.jar
> </code></pre>
> <p>And my pom.xml file</p>
> <pre><code class="lang-auto">&lt;?xml version="1.0" encoding="UTF-8"?&gt;
> &lt;project xmlns="http://maven.apache.org/POM/4.0.0"
>          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
>          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"&gt;
>     &lt;modelVersion&gt;4.0.0&lt;/modelVersion&gt;
> 
>     &lt;groupId&gt;onl.tesseract&lt;/groupId&gt;
>     &lt;artifactId&gt;tesseractlib&lt;/artifactId&gt;
>     &lt;version&gt;2.0-SNAPSHOT&lt;/version&gt;
>     &lt;packaging&gt;jar&lt;/packaging&gt;
> 
>     &lt;properties&gt;
>         &lt;java.version&gt;16&lt;/java.version&gt;
>         &lt;maven.compiler.source&gt;16&lt;/maven.compiler.source&gt;
>         &lt;maven.compiler.target&gt;16&lt;/maven.compiler.target&gt;
>         &lt;project.build.sourceEncoding&gt;UTF-8&lt;/project.build.sourceEncoding&gt;
>     &lt;/properties&gt;
>          
>     &lt;distributionManagement&gt;
>          &lt;repository&gt;
>              &lt;id&gt;github&lt;/id&gt;
>              &lt;name&gt;GitHub Packages&lt;/name&gt;
>              &lt;url&gt;https://maven.pkg.github.com/TesseractFR/TesseractLib&lt;/url&gt;
>          &lt;/repository&gt;
>      &lt;/distributionManagement&gt;
>          
>     &lt;build&gt;
>         &lt;plugins&gt;
>             &lt;plugin&gt;
>                 &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
>                 &lt;artifactId&gt;maven-compiler-plugin&lt;/artifactId&gt;
>                 &lt;version&gt;3.8.1&lt;/version&gt;
>                 &lt;configuration&gt;
>                     &lt;source&gt;16&lt;/source&gt;
>                     &lt;target&gt;16&lt;/target&gt;
>                 &lt;/configuration&gt;
>             &lt;/plugin&gt;
>             &lt;plugin&gt;
>                 &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
>                 &lt;artifactId&gt;maven-shade-plugin&lt;/artifactId&gt;
>                 &lt;version&gt;3.2.4&lt;/version&gt;
>                 &lt;executions&gt;
>                     &lt;execution&gt;
>                         &lt;phase&gt;package&lt;/phase&gt;
>                         &lt;goals&gt;
>                             &lt;goal&gt;shade&lt;/goal&gt;
>                         &lt;/goals&gt;
>                         &lt;configuration&gt;
>                             &lt;createDependencyReducedPom&gt;false&lt;/createDependencyReducedPom&gt;
>                         &lt;/configuration&gt;
>                     &lt;/execution&gt;
>                 &lt;/executions&gt;
>             &lt;/plugin&gt;
>             &lt;plugin&gt;
>                 &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
>                 &lt;artifactId&gt;maven-source-plugin&lt;/artifactId&gt;
>                 &lt;version&gt;3.2.1&lt;/version&gt;
>                 &lt;executions&gt;
>                     &lt;execution&gt;
>                         &lt;id&gt;attach-sources&lt;/id&gt;
>                         &lt;phase&gt;verify&lt;/phase&gt;
>                         &lt;goals&gt;
>                             &lt;goal&gt;jar-no-fork&lt;/goal&gt;
>                         &lt;/goals&gt;
>                     &lt;/execution&gt;
>                 &lt;/executions&gt;
>             &lt;/plugin&gt;
>         &lt;/plugins&gt;
>         &lt;resources&gt;
>             &lt;resource&gt;
>                 &lt;directory&gt;src/main/resources&lt;/directory&gt;
>                 &lt;filtering&gt;true&lt;/filtering&gt;
>             &lt;/resource&gt;
>         &lt;/resources&gt;
>     &lt;/build&gt;
> 
>     &lt;repositories&gt;
>         &lt;repository&gt;
>             &lt;id&gt;papermc&lt;/id&gt;
>             &lt;url&gt;https://papermc.io/repo/repository/maven-public/&lt;/url&gt;
>         &lt;/repository&gt;
>     &lt;/repositories&gt;
> 
>     &lt;dependencies&gt;
>         &lt;dependency&gt;
>             &lt;groupId&gt;io.papermc.paper&lt;/groupId&gt;
>             &lt;artifactId&gt;paper-api&lt;/artifactId&gt;
>             &lt;version&gt;1.17-R0.1-SNAPSHOT&lt;/version&gt;
>             &lt;scope&gt;provided&lt;/scope&gt;
>         &lt;/dependency&gt;
>         &lt;dependency&gt;
>             &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
>             &lt;artifactId&gt;maven-source-plugin&lt;/artifactId&gt;
>             &lt;version&gt;3.2.1&lt;/version&gt;
>             &lt;type&gt;maven-plugin&lt;/type&gt;
>         &lt;/dependency&gt;
>     &lt;/dependencies&gt;
> &lt;/project&gt;
> 
> </code></pre>
> <p>Any help would be appreciated</p>
